event-processing/lambda/cleanser-app.ts

```diff
@@ -5,30 +5,50 @@ import {
     FirehoseTransformationResult,
     FirehoseTransformationResultRecord,
 } from 'aws-lambda';
-import { ICleansedEvent } from './models/cleansed-event';
+import { HmacKeysEnum } from './enums/hmac-key.enum';
 import { EnrichedAuditEvent, IEnrichedAuditEvent } from './models/enriched-audit-event';
 import { CleansingService } from './services/cleansing-service';
+import { getHmacKey } from './services/key-service';
+import { ObfuscationService } from './services/obfuscation-service';
 
 export const handler = async (event: FirehoseTransformationEvent): Promise<FirehoseTransformationResult> => {
     /* Process the list of records and transform them */
-    const transformationResult: FirehoseRecordTransformationStatus = 'Ok';
+    let transformationResult: FirehoseRecordTransformationStatus = 'Ok';
+    let hmacKey = '';
+
+    try {
+        hmacKey = await getHmacKey(HmacKeysEnum.performance);
+    } catch (e) {
+        transformationResult = 'ProcessingFailed';
+        console.log('An error occurred getting the cleanser hmac key.  Failed with ' + e);
+    }
 
     const output = event.records.map((record: FirehoseTransformationEventRecord) => {
         const plaintextData: string = Buffer.from(record.data, 'base64').toString('utf-8');
         const events: unknown[] = JSON.parse(plaintextData);
-        const cleansedEvents: ICleansedEvent[] = [];
+        const cleansedEvents: unknown[] = [];
```

Comment:
Did this have to switch to "unknown"? I just did a little test and at least VScode didn't seem to mind?

Reply:
I was getting errors previously but have switched back and it seems ok, so will revert to ICleansedEvent[]